### ISSUE TITLE
inline_method ParameterSubstituter omits ConditionalAccess parent → SUCCESS + wrong semantics / CS0019 (?. binds only to right of ?: / ??)

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#1424]
## Summary

`inline_method` ParameterSubstituter omits `ConditionalAccessExpressionSyntax` from its `needsParens` parent check. When a parameter whose argument is a conditional (`?:`) or coalesce (`??`) expression is substituted into a `?.` receiver position, the rewriter drops required parentheses. Status = **SUCCESS**, but the result has **wrong semantics** and/or **CS0019**.

Peer `ParenthesizeIfNeeded` already parenthesizes for `MemberAccessExpressionSyntax` and also omits `ConditionalAccessExpressionSyntax` (same class of hole; separate from filed #1412 Invocation). This report focuses on ParameterSubstituter (sibling of filed #1417 Invocation / #1420 Cast).

## Master

`f78d23d2656831a408f04950d973ad240dd4a371`

### Before (compiles; prints `2`)

```csharp
using System;
class Program
{
    static string? Id(string? s) => s;
    static void Main()
    {
        bool pick = true;
        string? a = "hi";
        string? b = null;
        Console.WriteLine(Id(pick ? a : b)?.Length);
    }
}
```

### Tool

`inline_method` with `methodName=Id`, `removeMethod=true` (direct `InlineMethodOperation` / MCP equivalent).

### After (SUCCESS; prints `hi`)

```csharp
using System;
class Program
{
    static void Main()
    {
        bool pick = true;
        string? a = "hi";
        string? b = null;
        Console.WriteLine(pick ? a : b?.Length);
    }
}
```

`?.` binds only to `b`, so the ternary returns `string?` (`"hi"`) instead of `int?` (`2`).

Expected after: `Console.WriteLine((pick ? a : b)?.Length);` → still `2`.

### Before (compiles; prints `2`)

```csharp
using System;
class Program
{
    static string? Id(string? s) => s;
    static void Main()
    {
        string? a = null;
        string? b = "hi";
        Console.WriteLine(Id(a ?? b)?.Length);
    }
}
```

### After (SUCCESS; CS0019)

```csharp
Console.WriteLine(a ?? b?.Length);
```

`error CS0019: Operator '??' cannot be applied to operands of type 'string' and 'int?'`

Expected: `(a ?? b)?.Length`.

## CONTROL (Binary parent still parenthesizes)

`Inc(2 + 3)` with body `x + 1` → `(2 + 3) + 1` (ParameterSubstituter includes `BinaryExpressionSyntax`).

## Peer note (ParenthesizeIfNeeded also omits ConditionalAccess)

`Pick(true, "hi", null)?.Length` with body `pick ? a : b` → `true ? "hi" : null?.Length` → **CS0023**. CONTROL under `.Length` (MemberAccess) correctly yields `(true ? "hi" : "lo").Length`. Same omission in `ParenthesizeIfNeeded` ~L1429–1431; can be a follow-up bug if desired.

## Root cause

`src/RoslynMcp.Core/Refactoring/Inline/InlineMethodOperation.cs` — `ParameterSubstituter.VisitIdentifierName` ~L1454–1456:

```csharp
var needsParens = node.Parent is BinaryExpressionSyntax or MemberAccessExpressionSyntax
    or ConditionalExpressionSyntax or PrefixUnaryExpressionSyntax
    or ElementAccessExpressionSyntax or AssignmentExpressionSyntax or AwaitExpressionSyntax;
```

Omits `ConditionalAccessExpressionSyntax` (and already-filed Cast / Invocation). `MemberAccessExpressionSyntax` is present, so `.Length` is safe but `?.Length` is not.

Fix: add `or ConditionalAccessExpressionSyntax` to the `needsParens` parent check (and ideally the same for `ParenthesizeIfNeeded`).

## Distinct from

- #1418 — `inline_variable` needsParens omits ConditionalAccess/IsPattern (different op)
- #1417 — ParameterSubstituter Invocation parent
- #1420 — ParameterSubstituter Cast parent
- #1412 — ParenthesizeIfNeeded Invocation parent
- #1414 / #1419 — inline_variable unary/cast / await/switch

## Labels

`bug` only (open leftovers exist; not exclusive-end / leftover-0)

## Probe

`/workspace/fd-hunt-sep22-1906/` — `out-condaccess.txt`, runs under `runs/`
